sonaranalyzer-dotnet/src/Tests/SonarAnalyzer.UnitTest/TestCases/TestMethodShouldHaveCorrectSignature.cs

```diff
@@ -0,0 +1,105 @@
+using System;
+using Xunit;
+using NUnit.Framework;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Tests.Diagnostics
+{
+    class MsTestTest
+    {
+        [Microsoft.VisualStudio.TestTools.UnitTesting.TestMethod]
+        private void PrivateTestMethod() // Noncompliant {{Make this test method 'public'.}}
+//                   ^^^^^^^^^^^^^^^^^
+        {
+        }
+
+        [TestMethod]
+        protected void ProtectedTestMethod() // Noncompliant
+        {
+        }
+
+        [TestMethod]
+        internal void InternalTestMethod() // Noncompliant
+        {
+        }
+
+        [TestMethod]
+        public async void AsyncTestMethod()  // Noncompliant {{Make this test method non-'async'.}}
+        {
+        }
+
+        [TestMethod]
+        public void GenericTestMethod<T>()  // Noncompliant {{Make this test method non-generic.}}
+        {
+        }
+
+        [TestMethod]
+        private async void MultiErrorsMethod<T>() // Noncompliant {{Make this test method 'public', non-'async' and non-generic.}}
+        {
+        }
+
+        [TestMethod]
+        public async Task DoSomethingAsync() // Compliant
```

Comment:
XUnit, NUnit... I meant "add compliant tests for methods with attributes from XUnit and NUnit"